### PR TITLE
Team page BE, dirty

### PR DESCRIPTION
**Pull request by algyz200**
cia biski gali buti betvarke...

    Is Evaldo PR 26 rankiniu budu nukopijavau pakeistus FE failus
    Sukuriau nauja TeamControlleri kuris turi paduoti info TeamPage
    Kodas tam kontrolery yra ...... ojoj. bet veikia :D. Pataisysiu veliau, db nera laiko optimizavimams :)

Lukai, jei skaitai, tai uzsimerk ziuredamas i TeamControlleri. Pazadu, pataisysiu, db tiesiog svarbiausia kad info vaiksciotu

### CODE REVIEW

File: config/routes/annotations.yaml [hunk 1]
```diff
@@ -1,3 +1,7 @@
 controllers:
     resource: ../../src/Controller/
     type: annotation
+rest_controller:
+    resource: ../../src/Controller/Api
+    type: annotation
+    prefix: /api
```
Comment (zmoginas) [inline config/routes/annotations.yaml]: empty line

File: src/Controller/Api/TeamController.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+class TeamController extends AbstractFOSRestController
+{
+    /**
+     * List all Teams
+     * @Rest\Get("/teams", name="get_teams")
+     *
+     * @return Response
+     */
+
```
Comment (zmoginas) [inline src/Controller/Api/TeamController.php]: remove line

File: src/Controller/Api/TeamController.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    /**
+     * Create a new Team
+     * @Rest\Post("/team", name="post_team")
+     * @param Request $request
+     *
+     * @return Response
+     */
+
```
Comment (zmoginas) [inline src/Controller/Api/TeamController.php]: remove line

File: src/Controller/Api/TeamController.php [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+class TeamController extends AbstractFOSRestController
+{
+    /**
+     * List all Teams
+     * @Rest\Get("/teams", name="get_teams")
+     *
+     * @return Response
+     */
+
+    public function showTeams()
```
Comment (zmoginas) [inline src/Controller/Api/TeamController.php]: type hint

File: src/Controller/Api/TeamController.php [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+    /**
+     * Create a new Team
+     * @Rest\Post("/team", name="post_team")
+     * @param Request $request
+     *
+     * @return Response
+     */
+
+    public function createTeam(Request $request): Response
```
Comment (zmoginas) [inline src/Controller/Api/TeamController.php]: type hint

File: src/Controller/Api/TeamController.php [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+        return $this->handleView($this->view($form->getErrors()));
+    }
+
+    /**
+     * List all Teams
+     * @Rest\Get("/teams/{id}", name="get_teamsbyid")
+     *
+     * @return Response
+     */
+    public function getUsersByTeamId(int $id)
```
Comment (zmoginas) [inline src/Controller/Api/TeamController.php]: type hint

File: src/Controller/Form/TeamType.php [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+use App\Entity\Teams;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class TeamType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options)
```
Comment (zmoginas) [inline src/Controller/Form/TeamType.php]: type hint, annotations.

File: src/Controller/Form/TeamType.php [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        $builder
+            ->add('name', TextType::class)
+            ->add('githubRepo', TextType::class)
+            ->add('save', SubmitType::class);
+    }
+
+    public function configureOptions(OptionsResolver $resolver)
```
Comment (zmoginas) [inline src/Controller/Form/TeamType.php]: type hint, annotations.

File: src/Repository/ProjectsRepository.php [hunk 9]
```diff
@@ -19,6 +19,18 @@ public function __construct(ManagerRegistry $registry)
...
+//        $query = $this->createQueryBuilder('projects');
+//        $query
+//            ->select('projects')
+//        ->where($query->expr()->orX($query->expr()->eq('projects.id', $id)));
+//
+//
+//
+//        return $query->getQuery()->getArrayResult();
+//    }
+
```
Comment (zmoginas) [inline src/Repository/ProjectsRepository.php]: if you are commenting something, please remove before you pushing into to the develop or master

File: src/Repository/TeamsRepository.php [hunk 10]
```diff
@@ -19,6 +20,51 @@ public function __construct(ManagerRegistry $registry)
...
+     * @return array
+     */
+    public function getTeams(): array
+    {
+        $query = $this->createQueryBuilder('teams');
+        $query
+            ->select(
+                'teams'
+            );
+        return $query->getQuery()->getArrayResult();
```
Comment (zmoginas) [inline src/Repository/TeamsRepository.php]: blank between those two line

File: src/Repository/TeamsRepository.php [hunk 11]
```diff
@@ -19,6 +20,51 @@ public function __construct(ManagerRegistry $registry)
...
+     */
+
+    public function getTeamById(int $id)
+    {
+        $query = $this->createQueryBuilder('teams');
+        $query
+            ->select('teams')
+            ->where($query->expr()->orX($query->expr()->eq('teams.id', $id)));
+
+        return $query->getQuery()->getArrayResult();
```
Comment (zmoginas) [inline src/Repository/TeamsRepository.php]: blank between those two line

File: src/Repository/TeamsRepository.php [hunk 12]
```diff
@@ -19,6 +20,51 @@ public function __construct(ManagerRegistry $registry)
...
+//        $qb = $this->createQueryBuilder('team');
+//        $qb
+//            ->select('team.name')
+//            ->innerJoin('team.projects', 'projects')
+//            ->where($qb->expr()->eq('team.id', ':team_id'))
+//            ->setParameter('team_id', $teamId);
+//
+//        return $qb->getQuery()->getArrayResult();
+//    }
+
```
Comment (zmoginas) [inline src/Repository/TeamsRepository.php]: if you are commenting something, please remove before you pushing into to the develop or master

File: src/Controller/Api/TeamController.php (hunk 5 above)
Comment (algyz200) [reply src/Controller/Api/TeamController.php]: kokio cia type hint reikia?

File: src/Controller/Api/TeamController.php (hunk 6 above)
Comment (algyz200) [reply src/Controller/Api/TeamController.php]: ir cia?

File: src/Repository/TeamsRepository.php (hunk 11 above)
Comment (algyz200) [reply src/Repository/TeamsRepository.php]: tai viena eilute jau palikta pries return, palikt dar viena?

Comment (zmoginas) [reply src/Repository/TeamsRepository.php]: aha

File: src/Controller/Api/TeamController.php (hunk 6 above)
Comment (zmoginas) [reply src/Controller/Api/TeamController.php]: yup

File: src/Controller/Form/TeamType.php (hunk 8 above)
Comment (algyz200) [reply src/Controller/Form/TeamType.php]: nzn situ atveju koks typehint?

Comment (zmoginas) [reply src/Controller/Form/TeamType.php]: ar šitame metode tu ką nors gražinit? :sweat_smile:

**Pull request closed without merging**